src/daemon/abrtd.c

```diff
@@ -435,7 +435,24 @@ static gboolean handle_signal_cb(GIOChannel *gio, GIOCondition condition, gpoint
 {
     uint8_t signo;
     gsize len = 0;
-    g_io_channel_read_chars(gio, (void*) &signo, 1, &len, NULL);
+    GError *error = NULL;
+    GIOStatus stat = g_io_channel_read_chars(gio, (void*) &signo, 1, &len, &error);
+    if (stat == G_IO_STATUS_ERROR)
+    {
+        /* An error occurred on the channel. Report it and quit. */
+        error_msg_and_die(_("Can't read from gio channel: '%s'"),
+                error ? error->message : "");
```

Comment:
Can the error ever be null if the status indicates there was an error?

Reply:
I believe it is possible, at least in theory. The way some checks in `g_io_channel_read_chars()` are [implemented](https://github.com/GNOME/glib/blob/c8692fffe0723aee2bc6710590b5e70ef841b5cc/glib/giochannel.c#L1992) doesn't seem to preclude `error` being `NULL` even after a failure – for instance, when the channel is not readable. (Which _should not_ happen, but we better be safe than segfault-sorry, right?)

Reply:
Yeah, that makes sense.